### PR TITLE
MAINT: Bump to 10.3

### PR DESCRIPTION
**Pull request by larsoner**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] ~~Bumped the build number (if the version is unchanged)~~
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

As with https://github.com/conda-forge/pyobjc-framework-coreservices-feedstock/pull/20, no idea why this one didn't auto-tick. Opening for 10.3 and locally ran `conda-smithy rerender -c auto`. I suggest to merge this then I'll make a follow-up PR to bump to 10.3.1.

### CODE REVIEW

**Comment by hoechenberger** [comment]
New project URL is https://github.com/ronaldoussoren/pyobjc

In case you feel like updating that

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+{% set version = "10.3" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: 3a75f38bb1d5d2cf6a0d3e92801b3510f32e96cf6443d81b9dd92a84d72eff0a
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name|lower }}-{{ version }}.tar.gz
```
Comment (larsoner) [inline recipe/meta.yaml]: FYI this is a relevant change -- the previous releases had `pyobjc-framework-FSEvents` in the sdist name but 10.3 and 10.3.1 sdists have `pyobjc-framework-fsevents`. All wheels have the capitalized version, and other packages like `pyobjc-framework-coreservices` kept caps in the sdist name, so it's probably a bug that the sdist is not capitalized.

Comment (larsoner) [reply recipe/meta.yaml]: ... argh, and underscores

Comment (hoechenberger) [reply recipe/meta.yaml]: That probably explains why the bot didn't pick up on the new release

**Review by hoechenberger — APPROVED** [review]

**Comment by hoechenberger** [comment]
Great!

**Pull request merged**